hermes/serializers.py

```diff
@@ -138,59 +138,246 @@ class Meta(BaseTargetSerializer.Meta):
         fields = BaseTargetSerializer.Meta.fields + ['messages',]
 
 
-def validate_date(date, date_format=None):
-    if date_format:
-        if 'jd' in date_format.lower():
-            try:
-                float(date)
-            except ValueError:
-                raise serializers.ValidationError({'date': _(f"Date: {date} does not parse. JD formatted dates must be a float value.")})
-        else:
-            try:
-                datetime.strptime(date, date_format)
-            except ValueError:
-                raise serializers.ValidationError({'date': _(f"Date: {date} does not parse based on provided date format: {date_format}.")})
-    else:
+def validate_date(date):
+    try:
+        float(date)
+    except ValueError:
         try:
             parse(date)
         except ValueError:
-            raise serializers.ValidationError({'date': _(f"Date: {date} does not parse with dateutil.parser.parse. Please specify a date_format or change your date.")})
+            raise serializers.ValidationError({'date': _(f"Date: {date} does not parse with dateutil.parser.parse. Please specify the date in a standard format or as a JD.")})
 
 
-class HermesMessageSerializer(serializers.Serializer):
-    title = serializers.CharField(required=True)
-    topic = serializers.CharField(required=True)
-    message_text = serializers.CharField(required=False, default='', allow_blank=True)
-    submitter = serializers.CharField(required=True)
-    authors = serializers.CharField(required=False, default='', allow_blank=True)
+class ReferenceDataSerializer(serializers.Serializer):
+    source = serializers.CharField(required=False)
+    citation = serializers.CharField(required=False)
+    url = serializers.URLField(required=False)
+
+    def validate(self, data):
+        validated_data = super().validate(data)
+        if not (validated_data.get('source') and validated_data.get('citation')):
+            if validated_data.get('source'):
+                raise serializers.ValidationError({
+                    'citation': 'Must set citation with source'
+                })
+            if validated_data.get('citation'):
+                raise serializers.ValidationError({
+                    'source': 'Must set source with citation'
+                })
+            if not validated_data.get('url'):
+                raise serializers.ValidationError({
+                    'source': 'Must set source/citation or url',
+                    'citation': 'Must set source/citation or url',
+                    'url': 'Must set source/citation or url',
+                })
+        return validated_data
 
+class OrbitalElementsSerializer(serializers.Serializer):
+    epoch_of_elements = serializers.CharField(required=True)
+    orbinc = serializers.FloatField(required=True)
```

Comment:
Should we validate on proper angles?

Reply:
Feels like this whole section should have the same validation that we have in the obs portal, unless that's done elsewhere.

Reply:
Oh, never mind. I got bamboozled by the old `GenericHermesMessageSerializer` class

Reply:
sure we can do that... Do you just mean min/max value validation on each of the fields like we have? Since we don't have a type, we are just inferring what set of fields are required or not based on what you had said in the document right now.

Reply:
yeah, basically just min-maxes that make sense.
Inclination between 0 and 180,
eccentricity, and all the distances > 0
epoch of elements > 10000 
that sort of thing.